File: Program39.cpp
#include<iostream>
using namespace std;

class Number
{
    private:
        int iNo;
    
    public:
      Number(int iValue)
      {
          this->iNo = iValue;
      }
    public:
        bool CheckPrime()
        {
            int iCnt = 0;
            bool bFlag = true;

            for(iCnt = 2; iCnt <= iNo/2; iCnt++)
            {
                if((iNo % 2) == 0)
                {
                    bFlag = false;
                    break;
                }
            }
            return bFlag;
        }

};

int main()
{
    int iValue = 0;
    bool bRet;

    cout<<"Enter number"<<endl;
    cin>>iValue;
    
    Number nobj(iValue);

    bRet = nobj.CheckPrime();
    
    if(bRet == true)
    {
        cout<<iValue;
        cout<<" is a prime number";
    }
    else
    {
        cout<<iValue;
        cout<<" is not a prime number";
    }

    return 0;
}

File: Program39.java
import java.lang.*;
import java.util.*;

class Factors
{
    private int iNo;

    public void Accept()
    {
        Scanner sobj = new Scanner(System.in);
        System.out.println("Enter number");
        this.iNo = sobj.nextInt();
    }
    public void Display()
    {
        System.out.println("Entered value is:"+this.iNo);
    }
    public boolean CheckPrime()
    {
        int iCnt = 0;
        boolean bFlag = true;

        for(iCnt = 2; iCnt <= (iNo/2); iCnt++) 
        {
            if((iNo % iCnt) == 0)
            {
                bFlag = false;
                break;
            }
        }
        return bFlag;
        
    }
}

class Program39
{
    public static void main(String arg[])
    {
        boolean bRet;

        Factors fobj = new Factors();
        fobj.Accept();
        fobj.Display();
        bRet = fobj.CheckPrime();

        if(bRet == true)
        {
            System.out.println("Prime number");
        }
        else
        {
            System.out.println("Not prime number");
        }
    }
}